assets/src/edit-story/components/movable-element/index.js

```diff
@@ -0,0 +1,156 @@
+/**
+ * External dependencies
+ */
+import Moveable from 'react-moveable';
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback, useState, useRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useStory } from '../../app';
+import { getComponentForType } from '../../elements';
+
+const Element = styled.div`
+	cursor: pointer;
+	user-select: none;
+`;
+
+const MovableElement = ( props ) => {
```

Comment:
This is unrelated to the question of a single moveable vs many moveables below. But I just wanted to note that if we have a concept of "movable/draggable/panable/resizable/rotatable" - they are dependent on the component's states. E.g. an image is "movable", but when it's in the "background/fullbleed" state, it stops being "movable". Not sure ref forwarding would work well for that.